src/uwtools/api/config.py

```diff
@@ -124,17 +124,14 @@ def realize(
     total: bool = False,
     dry_run: bool = False,
     stdin_ok: bool = False,
-) -> None:
+) -> dict:
     """
     NB: This docstring is dynamically replaced: See realize.__doc__ definition below.
     """
-    input_config = (
-        _YAMLConfig(config=input_config) if isinstance(input_config, dict) else input_config
```

Comment:
It's no longer necessary to wrap a `dict` input in a `YAMLConfig` object, as `dict` objects are now accepted and handled correctly in the library.